src/rust/engine/graph/src/entry.rs

```diff
@@ -451,7 +451,7 @@ impl<N: Node> Entry<N> {
             generation,
             previous_result,
           }
-        } else if dirty {
+        } else if dirty && self.node.cacheable() {
```

Comment:
I think that since you now avoid dirtying uncacheable nodes, you can remove this?

Reply:
As I mention in the [NOTE comment](https://github.com/pantsbuild/pants/pull/9452/files#diff-43a9096a476afb7dee1a86d57f7f1176R1108-R1111), we still do dirty the uncacheable nodes. We could change it to not invalidate them, then we could remove this.